lalrpop/src/build/mod.rs

```diff
@@ -395,11 +395,24 @@ fn emit_recursive_ascent(session: &Session, grammar: &r::Grammar, report_file :
                 try!(lr1::codegen::test_all::compile(&grammar, user_nt, start_nt, &states, &mut rust)),
         }
 
-        rust!(rust,
-              "pub use self::{}parse{}::parse_{};",
-              grammar.prefix,
-              start_nt,
-              user_nt);
+        match grammar.nonterminals[&user_nt].visibility {
+            pt::Visibility::Priv => panic!(),
+            pt::Visibility::Pub(Some(ref path)) => {
+                rust!(rust,
+                      "pub({}) use self::{}parse{}::parse_{};",
+                      path,
+                      grammar.prefix,
+                      start_nt,
+                      user_nt);
+            }
+            pt::Visibility::Pub(None) => {
+                rust!(rust,
+                      "pub  use self::{}parse{}::parse_{};",
+                      grammar.prefix,
+                      start_nt,
+                      user_nt);
+            }
+        }
```

Comment:
The behavior of this section seems to just output `pub use` rather than `pub(crate) use`. Does anyone have any idea why? Did I get the pattern matching wrong?

Reply:
The `pub use` in question here seem to be emitted in `test_all.rs` https://github.com/lalrpop/lalrpop/blob/a527fa2ea232daf174370d4de90f3a49bfb8766c/lalrpop/src/lr1/codegen/test_all.rs#L58-L72

The reset of the `pub(crate)`s appear to be emitting correctly!